### PR TITLE
Added tests, removed unreachable code to get BRAVO to 100% coverage

### PR DESCRIPTION
**Pull request by umbernhard**
**Description**

Adds testing and removed unreachable logic to get BRAVO tests to 100% coverage. 

**Testing**

Tweaked some tests to cover corner cases. 

**Progress**

Ready to go.

### CODE REVIEW

**Review by benadida — APPROVED** [review]
looks ok to me, but recommend a @jonahkagan review on this one, too.

File: server/audit_math/bravo.py [hunk 1]
```diff
@@ -342,6 +344,12 @@ def get_sample_size(
 
     samples: Dict = {}
 
+    if round_sizes:
+        num_sampled = sum([round_sizes.get(rnd, 0) for rnd in round_sizes])
+        # If we've already sampled all the ballots, we should never be here
```
Comment (jonahkagan) [inline server/audit_math/bravo.py:349]: if we should never be here, shouldn't we just raise an exception?

File: server/audit_math/bravo.py [hunk 2]
```diff
@@ -374,7 +382,7 @@ def get_sample_size(
             best_loser = loser
 
     # If we're in a single-candidate race, set sample to 0
-    if margin["losers"] == 0:
+    if not margin["losers"]:
```
Comment (jonahkagan) [inline server/audit_math/bravo.py:385]: why this change?

File: server/audit_math/bravo.py [hunk 3]
```diff
@@ -432,6 +423,7 @@ def get_sample_size(
     if (
         num_ballots > large_election_threshold
         and samples["0.9"]["size"] >= all_ballots_threshold
+        or p_w == p_l
```
Comment (jonahkagan) [inline server/audit_math/bravo.py:426]: i think this might error in case that `p_w == p_l` because `samples` won't have key `"0.9"` set.

File: server/tests/audit_math/test_bravo.py [hunk 4]
```diff
@@ -598,3 +672,53 @@ def test_tied_contest():
         "all-ballots": {"type": "all-ballots", "size": 4992420, "prob": None},
     },
 }
+
+
+round2_sample_sizes = {
+    "test1": {
+        "asn": {"type": "ASN", "size": -12, "prob": 0.0},
```
Comment (jonahkagan) [inline server/tests/audit_math/test_bravo.py:679]: what's with all these negative sample sizes?

File: server/audit_math/bravo.py (hunk 1 above)
Comment (umbernhard) [reply server/audit_math/bravo.py:349]: I can do that!

File: server/audit_math/bravo.py (hunk 2 above)
Comment (umbernhard) [reply server/audit_math/bravo.py:385]: `margin["losers"]` contains an object, it's not a number.

File: server/audit_math/bravo.py (hunk 3 above)
Comment (umbernhard) [reply server/audit_math/bravo.py:426]: Hmm, it doesn't in the tests, but that's an interesting point. I did this to get around the linter warning that I was returning too many times.

File: server/tests/audit_math/test_bravo.py (hunk 4 above)
Comment (umbernhard) [reply server/tests/audit_math/test_bravo.py:679]: It's a quirk of the math. I guess we could be rounding them to zero, but these are cases that should never really be seen by Arlo....
